sdk/offchain/src/identifier.rs

```diff
@@ -107,30 +106,41 @@ pub enum IntentIdentifierError {
     Url(#[from] url::ParseError),
 }
 
-trait ToHRP {
-    /// Returns the hrp (human readable prefix) as defined in
-    /// https://dip.diem.com/dip-5/#format
-    fn to_hrp(&self) -> Result<String>;
+/// Defines the available HRPs (human readable prefix) as defined in
+/// https://dip.diem.com/dip-5/#format
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Ord, PartialOrd, Hash)]
+pub enum HumanReadablePrefix {
+    DM,
+    PDM,
+    TDM
 }
 
-impl ToHRP for ChainId {
-    fn to_hrp(&self) -> Result<String> {
-        let named_chain = NamedChain::from_chain_id(self);
-        let hrp = match named_chain {
-            Ok(NamedChain::MAINNET) => String::from("dm"),
-            Ok(NamedChain::PREMAINNET) => String::from("pdm"),
-            Ok(NamedChain::TESTNET) => String::from("tdm"),
-            Ok(NamedChain::DEVNET) => String::from("tdm"),
-            Ok(NamedChain::TESTING) => String::from("tdm"),
-            Err(error_str) => return Err(anyhow!(error_str)),
-        };
-        Ok(hrp)
+impl std::fmt::Display for HumanReadablePrefix {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            HumanReadablePrefix::DM => write!(f, "dm"),
+            HumanReadablePrefix::PDM => write!(f, "pdm"),
+            HumanReadablePrefix::TDM => write!(f, "tdm"),
+        }
+    }
+}
```

Comment:
In addition to this lets add an `as_str()` method so that we don't have to allocate by calling `to_string` when passing this to `bech32::encode`

```
fn as_str(&self) -> &str {
    match self {
        // ...
    }
}
```